src/runner/test.rs

```diff
@@ -73,31 +79,91 @@ fn run_cargo<DB: WriteResults>(
     };
 
     let mut did_ice = false;
-    let mut tracker = |line: &str| {
-        did_ice |= line.contains("error: internal compiler error");
+    let mut error_codes = BTreeSet::new();
+    let mut deps = BTreeSet::new();
+
+    let local_packages_id = build_env
+        .cargo()
+        .args(&["metadata", "--no-deps", "--format-version=1"])
+        .log_output(false)
+        .run_capture()?
+        .stdout_lines()
+        .iter()
+        .filter_map(|line| serde_json::from_str::<Metadata>(line).ok())
+        .flat_map(|metadata| metadata.packages.into_iter().map(|pkg| pkg.id))
+        .collect::<HashSet<_>>();
```

Comment:
The `run_cargo` function could be called multiple times for each task (for example, it's called two times for `test` -- first `cargo build` and then `cargo test`). Could we move this somewhere up the call stack?